File: CountLines/count_lines_3.py
#!/bin/env python
# -*- coding: utf-8 -*-

import sys

def count_lines( file_path ) :
  counter = 0
  with open( file_path, 'r' ) as file :
    while True :
      buffer = file.read( 4096 )
      if buffer :
        counter += buffer.count( '\n' )
      else :
        break

  return counter

if __name__ == '__main__' :

  import doctest
  doctest.testmod()

  print count_lines( sys.argv[ 1 ] )

File: CountLines/count_lines_3.pl
#!/bin/env perl

use strict;
use warnings;

# http://learn.perl.org/faq/perlfaq5.html#How-do-I-count-the-number-of-lines-in-a-file
sub count_lines {
  my $filename = shift;

  open( FILE, $filename );
  my $lines = 0;
  while ( read FILE, my $buffer, 4096 ) {
    $lines += ( $buffer =~ tr/\n// );
  }
  close FILE;

  return $lines;
}

print count_lines( $ARGV[ 0 ] );
